File: field/contests/atcoder/abc350/e.py
import sys
input = lambda: sys.stdin.readline().rstrip()
INF = 10**18

import sys,pypyjit
sys.setrecursionlimit(10**7)
pypyjit.set_param('max_unroll_recursion=0')
from functools import lru_cache

@lru_cache(maxsize=100000)
def dfs(n):
    if n == 0:
        return 0
    res1 = dfs(n//A) + X
    res2 = (dfs(n//2)+dfs(n//3)+dfs(n//4)+dfs(n//5)+dfs(n//6))/5 + Y*6/5
    return min(res1,res2)

N,A,X,Y = map(int,input().split())
print(dfs(N))

File: field/contests/atcoder/abc350/src/bin/e.rs
#![allow(dead_code,unused_imports,unused_variables,non_snake_case, non_upper_case_globals, path_statements)]
use memoise::memoise_map;
use proconio::{fastout, input,marker::{Chars, Bytes, Isize1, Usize1}};
use std::{
    cmp::{max, min, Reverse},
    collections::{BTreeMap, BTreeSet, BinaryHeap, HashMap, HashSet, VecDeque},
    mem::swap,
    process::exit,
};
use itertools::Itertools;
use superslice::Ext;
use rustc_hash::FxHashMap;

const INF: usize = 1 << 60;

// メモ化再帰
#[memoise_map(n)]
fn dfs(n:u64,A:u64,X:f64,Y:f64) -> f64{
    if n == 0 {
        return 0.0;
    }
    let res1 = dfs(n/A,A,X,Y) + X;
    let res2 = (2..=6).map(|i| dfs(n/i,A,X,Y)).sum::<f64>()/5.0 + Y*6.0/5.0;
    let res = res1.min(res2);
    return res;
}

#[fastout]
fn main() {
    input! {
        N: u64,
        A: u64,
        X: f64,
        Y: f64,
    }

    let ans = dfs(N,A,X,Y);

    println!("{}", ans);

}
